xtask/src/builder.rs

```diff
@@ -59,22 +60,29 @@ impl CrateSpec {
             CrateSpec::Local(s, _xip) => Some(s.to_string()),
             CrateSpec::CratesIo(n, _v, _xip) => Some(n.to_string()),
             CrateSpec::Prebuilt(n, _u, _xip) => Some(n.to_string()),
-            CrateSpec::BinaryFile(path, _xip) => Some(path.to_string()),
+            CrateSpec::BinaryFile(n, path, _xip) => {
+                if let Some(name) = n {
+                    Some(name.to_string())
+                }
+                else {
+                    Some(path.to_string())
+                }
+            },
             _ => None,
         }
     }
 }
-impl Clone for CrateSpec {
-    fn clone(&self) -> CrateSpec {
-        match self {
-            CrateSpec::Local(s, xip) => CrateSpec::Local(s.to_string(), *xip),
-            CrateSpec::CratesIo(n, v, xip) => CrateSpec::CratesIo(n.to_string(), v.to_string(), *xip),
-            CrateSpec::Prebuilt(n, u, xip) => CrateSpec::Prebuilt(n.to_string(), u.to_string(), *xip),
-            CrateSpec::BinaryFile(path, xip) => CrateSpec::BinaryFile(path.to_string(), *xip),
-            CrateSpec::None => CrateSpec::None,
-        }
-    }
-}
+//impl Clone for CrateSpec {
```

Comment:
Since this is not used, just delete it...we can find it in the git history if it has to be used again.